contracts/modules/LoanClosingsBase.sol

```diff
@@ -356,32 +356,55 @@ contract LoanClosingsBase is LoanClosingsShared, LiquidationHelper {
 		uint256 rolloverReward = _getRolloverReward(loanParamsLocal.collateralToken, loanParamsLocal.loanToken, loanLocal.principal);
 
 		if (rolloverReward != 0) {
-			// pay out reward to caller
-			loanLocal.collateral = loanLocal.collateral.sub(rolloverReward);
+			// if the reward > collateral:
+			if (rolloverReward > loanLocal.collateral) {
+				// 1. pay back the excess reward to the lender
+				// 2. pay the complete collateral to msg.sender
+				// 3. close the position & emit close event
+				uint256 excessRewards = rolloverReward.sub(loanLocal.collateral);
```

Comment:
this does not make sense. if the rollover reqwrad is 100 and the collateral is 90, why would you just close 10? You need to close the complete thing.